packages/web3/src/address/address.test.ts

```diff
@@ -17,51 +17,124 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
+      expect(() => validateAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kj')).not.toThrow()
+    })
+
+    it('should throw error for empty address', () => {
+      expect(() => validateAddress('')).toThrow('Address is empty')
+    })
+
+    it('should throw error for invalid base58 string', () => {
+      expect(() => validateAddress('InvalidBase58!!')).toThrow('Non-base58 character')
+    })
+
+    it('should throw error for invalid address type', () => {
+      const invalidAddressType = 0x04
+      const hash = randomBytes(32)
+      const bytes = Buffer.concat([Buffer.from([invalidAddressType]), hash])
+      const address = bs58.encode(bytes)
+      expect(() => validateAddress(address)).toThrow(`Invalid address: ${address}`)
+    })
+
+    it('should throw error for invalid multisig address', () => {
+      const AddressType = 0x01
+      const invalidData = Buffer.from('1234567890abcdef', 'hex')
+      const bytes = Buffer.concat([Buffer.from([AddressType]), invalidData])
+      const address = bs58.encode(bytes)
+      expect(() => validateAddress(address)).toThrow(`Invalid multisig address: ${address}`)
+    })
+
+    it('should throw error for invalid P2PKH address length', () => {
+      const addressType = 0x00
+      const invalidHashLength = 31
+      const hash = randomBytes(invalidHashLength)
+      const bytes = Buffer.concat([Buffer.from([addressType]), hash])
+      const address = bs58.encode(bytes)
+      expect(() => validateAddress(address)).toThrow(`Invalid address: ${address}`)
+    })
+
+    it('should throw error for invalid P2C address length', () => {
+      const addressType = 0x03
+      const invalidHashLength = 31
+      const hash = randomBytes(invalidHashLength)
+      const bytes = Buffer.concat([Buffer.from([addressType]), hash])
+      const address = bs58.encode(bytes)
+      expect(() => validateAddress(address)).toThrow(`Invalid address: ${address}`)
+    })
+
+    it('should throw error for invalid P2MPKH address with m <= 0', () => {
+      const n = 2
+      const mZero = 0
+      const bytesZero = lockupScriptCodec.encode({
+        kind: 'P2MPKH',
+        value: {
+          publicKeyHashes: [randomBytes(32), randomBytes(32)],
+          m: mZero
+        }
+      })
+      const addressZero = bs58.encode(bytesZero)
+      expect(() => validateAddress(addressZero)).toThrow(`Invalid multisig address, n: ${n}, m: ${mZero}`)
+
+      const mNegative = -1
+      const bytesNegative = lockupScriptCodec.encode({
+        kind: 'P2MPKH',
+        value: {
+          publicKeyHashes: [randomBytes(32), randomBytes(32)],
+          m: mNegative
+        }
+      })
+      const addressNegative = bs58.encode(bytesNegative)
+      expect(() => validateAddress(addressNegative)).toThrow(`Invalid multisig address, n: ${n}, m: ${mNegative}`)
+    })
+
+    it('should throw error for invalid P2MPKH address where n < m', () => {
+      const addressType = 0x01
```

Comment:
We can also use `lockupScriptCodec.encode` in this test, similar to how it is done in the test above.